src/settings/target/mod.rs

```diff
@@ -140,27 +156,48 @@ impl Manifest {
 
     pub fn generate(
         name: String,
-        target_type: TargetType,
+        target_type: Option<TargetType>,
         config_path: PathBuf,
         site: Option<Site>,
     ) -> Result<Manifest, failure::Error> {
+        let config_file = config_path.join("wrangler.toml");
+        let template_config_content = fs::read_to_string(&config_file);
+        let template_config = match &template_config_content {
+            Ok(content) => {
+                let config: TemplateConfig = toml::from_str(content)?;
+                if let Some(target_type) = target_type.clone() {
+                    if config.target_type != target_type {
+                        message::warn(&format!("The template recommends the \"{}\" type. Using type \"{}\" may cause errors, we recommend changing the type field in wrangler.toml to \"{}\"", config.target_type, target_type, config.target_type));
```

Comment:
ideally this edge case never even gets triggered, given that our templates should now have wrangler.toml in them!

Reply:
This case is for when users pass `--type` to `wrangler generate` and the template has a type specified in the toml

Reply:
The behavior is to override the template's toml with the type the user passed in and print this warning. Open to arguments saying you shouldn't be able to override

Reply:
Makes sense :)